src/core/ext/filters/client_channel/lb_policy/ring_hash/ring_hash.cc

```diff
@@ -646,8 +652,17 @@ void RingHash::RingHashSubchannelList::UpdateRingHashConnectivityStateLocked(
     state = GRPC_CHANNEL_TRANSIENT_FAILURE;
     start_connection_attempt = true;
   }
-  // Pass along status only in TRANSIENT_FAILURE.
-  if (state != GRPC_CHANNEL_TRANSIENT_FAILURE) status = absl::OkStatus();
+  // In TRANSIENT_FAILURE, report the last reported failure.
+  // Otherwise, report OK.
+  if (state == GRPC_CHANNEL_TRANSIENT_FAILURE) {
+    if (!status.ok()) {
+      last_failure_ = absl::UnavailableError(absl::StrCat(
+          "no reachable subchannels; last error: ", status.ToString()));
+    }
+    status = last_failure_;
+  } else {
+    status = absl::OkStatus();
```

Comment:
can a shutdown state reach here?

Reply:
No, subchannels never report SHUTDOWN.  The underlying ConnectedSubchannel does report SHUTDOWN, but in that case, the subchannel reports IDLE.

https://github.com/grpc/grpc/blob/86e282ba9792da83a11ac1ec03074ccc92e6d1aa/src/core/ext/filters/client_channel/subchannel.cc#L331